recipes/delete_validation.rb

```diff
@@ -21,8 +21,13 @@ class ::Chef::Recipe
   include ::Opscode::ChefClient::Helpers
 end
 
-file Chef::Config[:validation_key] do
-  action :delete
-  backup false
-  only_if { ::File.exists?(Chef::Config[:client_key]) }
+unless Chef::Config[:validation_key].nil?
+  file Chef::Config[:validation_key] do
+    action :delete
+    backup false
+    only_if {
+      ::File.exists?(Chef::Config[:validation_key]) &&
```

Comment:
You don't actually need to add this extra guard, since we want this resource to fire <i>iff</i> a client key exists.